app/src/Level2_simple_agentic_with_websearch.ipynb

```diff
@@ -0,0 +1,318 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Level 2: Simple Agentic \n",
+    "\n",
+    "This tutorial is designed to give new users an overview of how to use llama-stack's builtin agent framework effectively. It demonstrates basic single-tool usage and provides guidance on switching between a local development environment and a remote kubernetes cluster.\n",
+    "\n",
+    "## Overview\n",
+    "\n",
+    "This tutorial walks you through how to build your own AI agent who can search the web following these steps:\n",
+    "1. Connecting to a llama-stack server (remote or local).\n",
+    "2. Configuring an agent for tool use.\n",
+    "3. Running the agent.\n",
+    "\n",
+    "\n",
+    "## Prerequisites\n",
+    "\n",
+    "Before starting, ensure you have the following:\n",
+    "- Access to a remote cluster or a local Podman setup.\n",
+    "- API keys and user variables configured (e.g., `inference_model`, `TAVILY_SEARCH_API_KEY`, `LLAMA_STACK_ENDPOINT`).\n",
+    "- A Tavily API key is required. You can register for one at [https://tavily.com/](https://tavily.com/).\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Building a Web Search Agent\n",
+    "### 1. Connecting to a llama-stack server.\n",
+    "#### 1.1 Setting Up the Environment\n",
+    "- Import the necessary libraries.\n",
+    "- Defining user variables.\n",
+    "- Connection options\n",
+    "    - **Remote Server**: Set `remote=True` and provide the remote llama-stack endpoint.\n",
+    "    - **Local Server**: Set `remote=False` and provide the local llama-stack port (default: 8321).\n",
+    "    \n",
+    "    For detailed llama-stack server setup instructions, refer to:\n",
+    "    - [Remote Setup Guide](https://github.com/opendatahub-io/llama-stack-on-ocp/tree/main/kubernetes)\n",
+    "    - [Local Setup Guide](https://github.com/redhat-et/agent-frameworks/tree/main/prototype/frameworks/llamastack)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Model: ibm-granite/granite-3.2-8b-instruct\n"
+     ]
+    }
+   ],
+   "source": [
+    "from llama_stack_client.lib.agents.agent import Agent\n",
+    "from llama_stack_client.lib.agents.event_logger import EventLogger\n",
+    "from llama_stack_client import LlamaStackClient\n",
+    "from termcolor import cprint\n",
+    "\n",
+    "inference_model = \"ibm-granite/granite-3.2-8b-instruct\"\n",
+    "TAVILY_SEARCH_API_KEY = \"YOUR_TAVILY_API_KEY\" # Replace with your tavily API key\n",
```

Comment:
Just a general note: ALL_CAPS_VARIABLES are typically used for environment variables, and lower case names are used for standard variables. The way they are mixed here looks a little odd. Since we are not using any env variables, can we just make them all lower case?